project-addons/transportation/models/sale.py

```diff
@@ -38,6 +38,25 @@ def onchange_partner_id(self):
             self.service_id = self.partner_id.service_id.id
         return res
 
+    @api.multi
+    @api.onchange('partner_id')
+    def onchange_partner_id_transporter(self):
+        transporter_ids=list()
+        all_transporters=self.env['transportation.transporter'].search([])
+        partnerCountry=self.partner_id.country_id
+        for x in all_transporters:
+            if partnerCountry in x.country_group_id.country_ids:
```

Comment:
Esto podrías hacerlo en un domain y ahorras lineas: self.env['transportation.transporter'].search([('country_group_id.country_ids', 'in', [self.partner_id.country_id.id])

Reply:
Cambios hechos.